### ISSUE TITLE
Consider support yaml format for rule tests

### ISSUE THREAD

**Issue by puzrin** [eslint/eslint#1223]
Tests for rules are just json objects. IMHO, it would be more convenient if devs could use yaml to describe data. Especially, when test samples needs multiline strings.

Example:

``` yaml
indent:

  valid:
    - |
      var a
        , b;

    - |
      if (a > 0) {
        b++;
      }

  invalid:
    -
      code: |
        if (a > 0) {
             b++;
        }

      errors:
        - message: 'Bad indentation (5 instead 2).'
```

Of cause, it's possible to write it in more compact form, if needed.
